matplotlib_visual/graph.py

```diff
@@ -4,30 +4,54 @@
 
 
 def display(snippet_name):
+    """
+    Displays the contents of a Python code snippet file and copies the source code
+    to the clipboard based on the operating system.
+
+    Parameters:
+    snippet_name (str): The name of the code snippet file (without .py extension)
+                        to display and copy.
+
+    Raises:
+    OSError: If the operating system is unsupported for copying to the clipboard.
+    """
     snippet_path = os.path.join(
         os.path.dirname(__file__), "code_snippets", f"{snippet_name}.py"
     )
 
     with open(snippet_path, "r") as file:
         source_code = file.read()
 
-    # Function to copy text to clipboard based on platform
     def copy_to_clipboard(text):
-        # Linux
+        """
+        Copies the provided text to the clipboard based on the operating system.
+
+        Parameters:
+        text (str): The text to copy to the clipboard.
+
+        Raises:
+        OSError: If the operating system is unsupported for copying to the clipboard.
+        """
         if "linux" in sys.platform:
             subprocess.run(
                 ["/usr/bin/xclip", "-selection", "clipboard"],
                 input=text.strip().encode(),
                 check=True,
             )
-
-        # Windows
         elif "win32" in sys.platform:
             subprocess.run(
                 ["C:\\Windows\\System32\\clip.exe"],
                 input=text.strip().encode(),
                 check=True,
             )
-
+        elif "darwin" in sys.platform:
```

Comment:
I see that you added support for iOS, and I really appreciate your efforts! However, that was not the intended outcome. It was actually already fixed while I was rebasing with the previous branches. I’ll rename this PR and issue to reflect the addition of iOS support.